test/integrationtest/moduletest/commercesearch_test.go

```diff
@@ -0,0 +1,102 @@
+// +build integration
+
+package moduletest_test
+
+import (
+	"net/http"
+	"testing"
+
+	"flamingo.me/dingo"
+
+	"flamingo.me/flamingo/v3/framework/config"
+
+	"flamingo.me/flamingo-commerce/v3/test/integrationtest"
+
+	"flamingo.me/flamingo-commerce-adapter-standalone/commercesearch"
+	"flamingo.me/flamingo-commerce-adapter-standalone/csvindexing"
+)
+
+func testEmptySearchReturnsAllProducts(info integrationtest.BootupInfo) func(t *testing.T) {
+	return func(t *testing.T) {
+		t.Helper()
+		t.Parallel()
+		e := integrationtest.NewHTTPExpect(t, "http://"+info.BaseURL)
+		expect := e.GET("/search").Expect()
+		productsResult := expect.Status(http.StatusOK).JSON().Object().Value("SearchResult").Object().Value("products").Object()
+		productsResult.Value("Hits").Array().Length().Equal(50)
+		searchMeta := productsResult.Value("SearchMeta").Object()
+		searchMeta.Value("NumResults").Number().Equal(67)
+		searchMeta.Value("NumPages").Number().Equal(2)
+		searchMeta.Value("Page").Number().Equal(1)
+	}
+}
+
+func testSearchWithQueryReturnsAllMatchingProducts(info integrationtest.BootupInfo) func(t *testing.T) {
+	return func(t *testing.T) {
+		t.Helper()
+		t.Parallel()
+		e := integrationtest.NewHTTPExpect(t, "http://"+info.BaseURL)
+		expect := e.GET("/search").WithQueryString("q=Refrigerator").Expect()
+		productsResult := expect.Status(http.StatusOK).JSON().Object().Value("SearchResult").Object().Value("products").Object()
+		productsResult.Value("Hits").Array().Length().Equal(1)
```

Comment:
Should we check if the found hit is really the "Refrigerator"?

Reply:
Updated the test, please have a look 👍🏻